protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/encoding/BackupStatusResponse.java

```diff
@@ -117,13 +126,35 @@ public void wrap(final DirectBuffer buffer, final int offset, final int length)
     brokerId = bodyDecoder.brokerId();
     checkpointPosition = bodyDecoder.checkpointPosition();
     numberOfPartitions = bodyDecoder.numberOfPartitions();
-    snapshotId = bodyDecoder.snapshotId();
-    failureReason = bodyDecoder.failureReason();
+
+    encodedSnapshotId = new byte[bodyDecoder.snapshotIdLength()];
+    bodyDecoder.getSnapshotId(encodedSnapshotId, 0, encodedSnapshotId.length);
+    snapshotId =
+        decodeString(encodedSnapshotId, BackupStatusResponseDecoder.snapshotIdCharacterEncoding());
+
+    encodedFailureReason = new byte[bodyDecoder.failureReasonLength()];
+    bodyDecoder.getFailureReason(encodedFailureReason, 0, encodedFailureReason.length);
+    failureReason =
+        decodeString(
+            encodedFailureReason, BackupStatusResponseDecoder.failureReasonCharacterEncoding());
+  }
+
+  private String decodeString(final byte[] encodedSnapshotId, final String charsetName) {
+    try {
+      return new String(encodedSnapshotId, charsetName);
+    } catch (final UnsupportedEncodingException e) {
+      throw new RuntimeException(e);
+    }
   }
 
   @Override
   public int getLength() {
-    return 0;
+    return headerEncoder.encodedLength()
+        + bodyEncoder.sbeBlockLength()
+        + BackupStatusResponseEncoder.snapshotIdHeaderLength()
+        + encodedSnapshotId.length
```

Comment:
:question: Can't we just use bodyEncoder.encodedLength()? That way we also wouldn't need to maintain two variables (byte array + decoded string) each for snapshotId and failureReason.

Reply:
No. `write` may not have been called before `getLength`. bodyEncoder.encodedLength() is only correct after `write` is called.